src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -171,9 +171,8 @@ abstract sealed class MatrixIR extends BaseIR {
       .getOrElse(
         Optimize(
           TableMapRows(
-            MatrixRowsTable(this),
+            TableUnkey(MatrixRowsTable(this)),
```

Comment:
same question

Reply:
Again, because otherwise we would have to tell the `TableMapRows` the existing keys of the `MatrixRowsTable`, and since we don't care about keys here it's easier to just unkey first.

The `newKey` parameter of `TableMapRows` is only there to support renaming (but not otherwise changing) key fields. I don't really like it, but all the alternatives I could think of were even messier. Thoughts are welcome.